### ISSUE TITLE
Failed to find guest pc for host ptr `0x7fbb95d00e72`

### ISSUE THREAD

**Issue by jserv** [bane9/RISCVBox#1]
I managed to launch RISCVBox via the command `cargo r --release -- --bios linux/prebuilt/fw_jump.bin --kernel linux/prebuilt/Image` and got the error messages:
```
thread '<unnamed>' panicked at src/frontend/exec_core.rs:128:25:
Failed to find guest pc for host ptr 0x7fa8edc7de72
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Then, I set `RUST_BACKTRACE=full` and launched again:
```
stack backtrace:
   0:     0x565121d0694c - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::he01d6237f59aeb48
   1:     0x565121cae5fb - core::fmt::write::h1b2ecdc496e16450
   2:     0x565121cdd892 - std::io::Write::write_fmt::hcad97c424e3160f0
   3:     0x565121d0b758 - std::panicking::default_hook::{{closure}}::h4739b3309167c46a
   4:     0x565121d0c672 - std::panicking::rust_panic_with_hook::h883d4a74d97bf3d2
   5:     0x565121d0c165 - std::panicking::begin_panic_handler::{{closure}}::he3fd84ab1e1b81ad
   6:     0x565121d0c0c9 - std::sys::backtrace::__rust_end_short_backtrace::h1c973dd42a09963a
   7:     0x565121d0c0b4 - rust_begin_unwind
   8:     0x565121cad5c2 - core::panicking::panic_fmt::h9f12ebfa09d4e7f8
   9:     0x565121c48036 - RISCVBox::frontend::exec_core::exec_core_thread::hc330c3a4ae278eab
  10:     0x565121c5c186 - std::sys::backtrace::__rust_begin_short_backtrace::haf754f81e970c113
  11:     0x565121c36e3e - core::ops::function::FnOnce::call_once{{vtable.shim}}::hcec242091c719e33
  12:     0x565121d0d48b - std::sys::pal::unix::thread::Thread::new::thread_start::h522b9b1f732bad99
  13:     0x7f56194f2609 - start_thread
                               at /build/glibc-LcI20x/glibc-2.31/nptl/pthread_create.c:477:8
  14:     0x7f561965b353 - clone
                               at /build/glibc-LcI20x/glibc-2.31/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95
  15:                0x0 - <unknown>
```

Tested on Ubuntu Linux 20.04-LTS.

**Comment by bane9** [comment]
Hi Jim,

Currently the Linux build is broken, it will get fixed at some point but currently no ETA. It is recommended to use Windows with RISCVBox for now.

**Comment by bane9** [comment]
I was able to get it functioning in 725f32c, and in the latest build v1.2.6.
This is a temporary workaround as at this time I don't understand the discrepency between Windows and Linux when it comes to this issue.

Framebuffer console doesn't appear to be currently working on Linux (possibly a Wayland related issue?). 

Tested on Ubuntu 23.10 under VirtualBox

Can you please confirm it also works on your system by running it as:

cargo r --release -- --bios linux/prebuilt/fw_jump.bin --kernel linux/prebuilt/Image --nographic

**Comment by bane9** [comment]
Update: The framebuffer console is working under WSL using X Windows, though the exact cause remains unclear for the unstable behaviour on Ubuntu 23.10. It might be related to Wayland, the distribution, or possibly a GTK issue, I'm not sure.

**Comment by jserv** [comment]
> Can you please confirm it also works on your system by running it as: `cargo r --release -- --bios linux/prebuilt/fw_jump.bin --kernel linux/prebuilt/Image --nographic`

Yes, it works for me. I was able to get the following message and shell:
```
...
[    3.005000] Freeing unused kernel image (initmem) memory: 2612K
[    3.005000] Kernel memory protection not selected by kernel config.
[    3.006000] Run /init as init process
```

**Comment by bane9** [comment]
Further update, the issue is completely resolved in feed287
